testdata/p4_16_samples/psa-resubmit-bmv2.p4

```diff
@@ -0,0 +1,144 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <core.p4>
+#include "psa.p4"
+
+
+typedef bit<48>  EthernetAddress;
+
+header ethernet_t {
+    EthernetAddress dstAddr;
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+struct empty_metadata_t {
+}
+
+struct metadata_t {
+}
+
+struct headers_t {
+    ethernet_t       ethernet;
+}
+
+parser IngressParserImpl(packet_in pkt,
+                         out headers_t hdr,
+                         inout metadata_t user_meta,
+                         in psa_ingress_parser_input_metadata_t istd,
+                         in empty_metadata_t resubmit_meta,
+                         in empty_metadata_t recirculate_meta)
+{
+    state start {
+        pkt.extract(hdr.ethernet);
+        transition accept;
+    }
+}
+
+control cIngress(inout headers_t hdr,
+                 inout metadata_t user_meta,
+                 in    psa_ingress_input_metadata_t  istd,
+                 inout psa_ingress_output_metadata_t ostd)
+{
+
+    action resubmit() {
+        // this write should not make it to the output packet
+        hdr.ethernet.dstAddr = 48w256;
+        ostd.resubmit = true;
+    }
+
+    action pkt_write() {
+        ostd.drop = false;
+        hdr.ethernet.dstAddr = 48w4;
+    }
+
+    apply {
+        pkt_write();
```

Comment:
Thinking about this test case a bit more, it would be a better test if resubmit() and pkt_write() modified _different_ fields of the packet header.  Why?  Because as the test is written now, if psa_switch were incorrectly preserving modifications to the packet from before resubmitting, to after resubmitting, this test will not catch that problem.

For example, if resubmit() had the assignment `hdr.ethernet.srcAddr = 48w256;` _instead of_ the assignment to `hdr.ethernet.dstAddr`, then the test would catch a buggy psa_switch implementation that preserved that modification to srcAddr across the resubmit operation.

Reply:
True. My bad, should have caught this earlier.